rclcpp/include/rclcpp/parameter_map.hpp

```diff
@@ -19,18 +19,23 @@
 
 #include <string>
 #include <unordered_map>
+#include <utility>
 #include <vector>
 
 #include "rclcpp/exceptions.hpp"
 #include "rclcpp/parameter.hpp"
 #include "rclcpp/parameter_value.hpp"
 #include "rclcpp/visibility_control.hpp"
+#include "rcl_interfaces/msg/parameter_descriptor.hpp"
 
 namespace rclcpp
 {
 
 /// A map of fully qualified node names to a list of parameters
-using ParameterMap = std::unordered_map<std::string, std::vector<Parameter>>;
+using rcl_interfaces::msg::ParameterDescriptor;
+using ParameterAndDescriptor = std::unordered_map<std::string, std::pair<Parameter,
```

Comment:
@bpwilcox `std::pair` is convenient but obscure when reading code (e.g. what is [`param.second.second`](https://github.com/ros2/rclcpp/pull/909/files#diff-43c1c52da4c73f0a25534075d66ece63R117)??). Consider using defining a struct instead.

Reply:
I think it may be even easier to directly create the `ParameterInfo` struct here.